### ISSUE TITLE
'atommap' command unexpected result

### ISSUE THREAD

**Issue by juan-alcantara1** [Amber-MD/cpptraj#1074]
I have two mol2 files containing the same molecule (same number of atoms) but the atoms are not ordered the same way in the two files. I followed the example available for the _atommap_ cpptraj command  and it ran without error, but it incorrectly mapped some of the atoms even though they are named the same in both files. Aligning both structures in ChimeraX, it shows that equivalent atoms have the same name so It's odd that _atommap_ wasn't able to do the same matching.
 
[publishedHeme_reordered.txt](https://github.com/Amber-MD/cpptraj/files/15013290/publishedHeme_reordered.txt)
[6o3i_heme_wH.txt](https://github.com/Amber-MD/cpptraj/files/15013291/6o3i_heme_wH.txt)
[atommap.txt](https://github.com/Amber-MD/cpptraj/files/15013292/atommap.txt)
[cpptraj_atommap_log.log](https://github.com/Amber-MD/cpptraj/files/15013293/cpptraj_atommap_log.log)
[cpptraj_atommap.txt](https://github.com/Amber-MD/cpptraj/files/15013294/cpptraj_atommap.txt)
[publishedHeme.txt](https://github.com/Amber-MD/cpptraj/files/15013295/publishedHeme.txt)

**Comment by drroe** [comment]
Thanks for the report, I'll take a look.

**Comment by drroe** [comment]
OK - I think I have a fixed based on ordering bonded atoms via priority. I'm testing it now and hopefully will have it in production early next week.

**Comment by juan-alcantara1** [comment]
Thank you!

**Comment by drroe** [comment]
See if this structure seems better. The RMS to the 603i reference is 1.5971 Ang, which seems good. All the names match up except the following:
```
TgtAt  Tgt  RefAt  Ref
    64 HBA2     50 HBA1
    63 HBA1     51 HBA2
    67 HBD2     56 HBD1
    66 HBD1     57 HBD2
```
I visually inspected and it seems like the map provided by cpptraj does add up, and the names were actually switched (i.e. this is a real result). Let me know what you think.
[publishedHeme_reordered.mol2.txt](https://github.com/Amber-MD/cpptraj/files/15144135/publishedHeme_reordered.mol2.txt)

**Comment by drroe** [comment]
Here is the other flip (`603i_heme_wH.mol2` on the left, `publishedHeme.mol2` on the right):
![flipped2](https://github.com/Amber-MD/cpptraj/assets/10144183/a917defb-1743-4229-8944-6729c3b4c9dd)

**Comment by drroe** [comment]
Hopefully addressed by #1076.
